### PR TITLE
Add warning when using templates in service dev tools

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adds a warning to the service dev tools, when using templates:

PS: I have no idea what I'm doing 😬 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/developer-tools/service/developer-tools-service.ts [hunk 1]
```diff
@@ -195,6 +199,15 @@ class HaPanelDevService extends LitElement {
...
     }
+
+    this._error = undefined;
+    if (hasTemplate(this._serviceData!)) {
+      this._error = this.hass.localize(
+        "ui.panel.developer-tools.tabs.services.no_templates"
+      );
+      return false;
+    }
+
```
Comment (bramkragten) [inline src/panels/developer-tools/service/developer-tools-service.ts:210]:
```suggestion
    if (hasTemplate(this._serviceData!)) {
      this._error = this.hass.localize(
        "ui.panel.developer-tools.tabs.services.no_templates"
      );
      return false;
    }
    this._error = undefined;

```

File: src/panels/developer-tools/service/developer-tools-service.ts [hunk 2]
```diff
@@ -195,6 +199,15 @@ class HaPanelDevService extends LitElement {
     if (!serviceData?.service) {
       return false;
     }
+
+    this._error = undefined;
+    if (hasTemplate(this._serviceData!)) {
```
Comment (bramkragten) [inline src/panels/developer-tools/service/developer-tools-service.ts:204]: We should move this check down as it is pretty heavy compared to the rest

File: src/panels/developer-tools/service/developer-tools-service.ts [hunk 3]
```diff
@@ -195,6 +199,15 @@ class HaPanelDevService extends LitElement {
...
+      this._error = this.hass.localize(
```
Comment (bramkragten) [inline src/panels/developer-tools/service/developer-tools-service.ts:205]: A memoize should not have side effects, you could return the error:
`return {isValid: false, error: error};`
and it render:
`const {isValid, error} = this._isValid(this._serviceData, fields, target);`

File: src/panels/developer-tools/service/developer-tools-service.ts [hunk 4]
```diff
@@ -105,6 +105,7 @@ class HaPanelDevService extends LitElement {
                   @value-changed=${this._serviceDataChanged}
                 ></ha-service-control></div
             ></ha-card>`}
+        ${error ? html` <p class="error">Error: ${error}</p> ` : ""}
```
Comment (bramkragten) [inline src/panels/developer-tools/service/developer-tools-service.ts:108]: `Error` should be localized

**Review by bramkragten — APPROVED** [review]

File: src/translations/en.json [hunk 5]
```diff
@@ -3340,7 +3340,8 @@
             "ui_mode": "Go to UI mode",
             "yaml_parameters": "Parameters only available in YAML mode",
             "all_parameters": "All available parameters",
-            "accepts_target": "This service accepts a target, for example: `entity_id: light.bed_light`"
+            "accepts_target": "This service accepts a target, for example: `entity_id: light.bed_light`",
+            "no_templates": "The service dev tools does not support the use of Jinja templates"
```
Comment (spacegaier) [inline src/translations/en.json:3344]:
```suggestion
            "no_templates": "The service dev tool does not support the use of Jinja templates"
```

It is a single one of the four dev tools.

**Comment by frenck** [comment]
Closing this awaiting: https://github.com/home-assistant/core/pull/47804

**Pull request closed without merging**